dataconnectors/file/file.go

```diff
@@ -145,3 +150,33 @@ func (c *FileConnector) processWatchNotifyEvent(event fsnotify.Event, path strin
 
 	return nil
 }
+
+func (c *FileConnector) sendData() error {
+	if len(c.readHandlers) == 0 || c.fileInfo == nil || c.data == nil {
+		// Nothing to read
+		return nil
+	}
+
+	metadata := map[string]string{}
+	metadata["mod_time"] = c.fileInfo.ModTime().Format(time.RFC3339Nano)
```

Comment:
wonder if this would work on a Windows machine?

Reply:
it should, windows supports high resolution times.